src/transformers/models/mgp_str/configuration_mgp_str.py

```diff
@@ -0,0 +1,144 @@
+# coding=utf-8
+# Copyright 2022 Google AI and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" MGPSTR model configuration"""
+
+from ...configuration_utils import PretrainedConfig
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+MGP_STR_PRETRAINED_CONFIG_ARCHIVE_MAP = {
+    "alibaba-damo/mgp-str-base": "https://huggingface.co/alibaba-damo/mgp-str-base/resolve/main/config.json",
+}
+
+
+class MGPSTRConfig(PretrainedConfig):
+    r"""
+    This is the configuration class to store the configuration of a [`MGPSTRModel`]. It is used to instantiate an
+    MGPSTR model according to the specified arguments, defining the model architecture. Instantiating a configuration
+    with the defaults will yield a similar configuration to that of the MGPSTR
+    [alibaba-damo/mgp-str-base](https://huggingface.co/alibaba-damo/mgp-str-base) architecture.
+
+    Configuration objects inherit from [`PretrainedConfig`] and can be used to control the model outputs. Read the
+    documentation from [`PretrainedConfig`] for more information.
+
+
+    Args:
+        img_size (`int`, *optional*, defaults to `32x128`):
+            The size (resolution) of each image.
+        patch_size (`int`, *optional*, defaults to `4`):
+            The size (resolution) of each patch.
+        in_chans (`int`, *optional*, defaults to `3`):
+            The number of input channels.
+        max_token_length (`int`, *optional*, defaults to `27`):
+            The max number of output tokens.
+        char_num_classes (`int`, *optional*, defaults to `27`):
+            The number of classes for char head .
+        bpe_num_classes (`int`, *optional*, defaults to `27`):
+            The number of classes for bpe head .
+        wp_num_classes (`int`, *optional*, defaults to `27`):
+            The number of classes for wp head .
+        embed_dim (`int`, *optional*, defaults to 768):
+            The embedding dimension.
+        depth (`int`, *optional*, defaults to 12):
+            Number of hidden layers in the Transformer encoder.
```

Comment:
```suggestion
        num_hidden_layers (`int`, *optional*, defaults to 12):
            Number of hidden layers in the Transformer encoder.
```
Let's use the same terminology as other models in the library